File: corelibrary/src/main/java/pers/cxd/corelibrary/base/OnActivityResultCallback.java
package pers.cxd.corelibrary.base;

import android.content.Intent;

import androidx.annotation.Nullable;

/**
 * A callback support to {@link UiComponent#registerActivityResultCallback(OnActivityResultCallback)}
 *
 * @author pslilysm
 * @since 1.0.4
 */
public interface OnActivityResultCallback {

    void onActivityResult(int requestCode, int resultCode, @Nullable Intent data);

}

File: corelibrary/src/main/java/pers/cxd/corelibrary/base/OnActivityResultCallback.kt
package pers.cxd.corelibrary.base

import android.content.Intent

/**
 * A callback support to [UIComponent.registerActivityResultCallback]
 *
 * @author pslilysm
 * @since 1.0.4
 */
interface OnActivityResultCallback {
    fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?)
}